theano/gof/destroyhandler.py

```diff
@@ -808,13 +808,20 @@ def fast_destroy(self, app):
             destroyed_inputs_idx += i
         for d_inp_idx in destroyed_inputs_idx:
             d_inp = app.inputs[d_inp_idx]  # The destroyed input variable
-            if len(d_inp.clients) == 1:
-                # We are fine, the algo tell we shouldn't generate cycle
-                continue
-            # Current simplest version don't allow destroyed inputs to
-            # have more then 1 client.
+            if len(d_inp.clients) != 1:
+                # We allow only 1 clients on variable we destroy
                 self.fail_validate = True
-                
+
+            # If the variable destroyed isn't a view, we are fine.
+            if not d_inp.owner or not getattr(d_inp.owner.op,
+                                              'view_map', False):
+                continue
+            # The node have views, verify if this specific output is a view.
```

Comment:
right.

On Tue, May 17, 2016 at 12:59 PM, Ramana Subramanyam <
notifications@github.com> wrote:

> In theano/gof/destroyhandler.py
> <https://github.com/Sentient07/Theano/pull/1#discussion_r63561610>:
>
> >                  self.fail_validate = True
> > -
> > +
> > +            # If the variable destroyed isn't a view, we are fine.
> > +            if not d_inp.owner or not getattr(d_inp.owner.op,
> > +                                              'view_map', False):
> > +                continue
> > +            # The node have views, verify if this specific output is a view.
>
> One small clarification. If a variable doesn't have an owner, it is an
> input right ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Sentient07/Theano/pull/1/files/ba89cefc786a7a20ac0a028e27d40d081cd29915#r63561610>
>